cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -287,9 +287,9 @@ function enableInstantSearch(config) {
                                         <div class="price">
                                             ${isPricingLazyLoad && html`<span class="price-placeholder" data-product-id="${productId}"></span>`}
                                             ${!isPricingLazyLoad && html`
-                                                ${ (hit.displayPrice < hit.price || (hit.promotionalPrice && hit.promotionalPrice < hit.price)) && html`
+                                                ${ (hit.displayPrice < hit.price || (hit.promotionalPrice && hit.promotionalPrice < hit.price) || (hit.highestPriceBookPrice && hit.highestPriceBookPrice > hit.displayPrice)) && html`
```

Comment:
You've made things more clear with the `calculateDisplayPrice` function, but you have complexified this part.
It's a good occasion to improve the `transformItem` function and generate prices ready to be displayed:
- a `strikeout` price if there is one
- a `displayPrice`

So the HTML doesn't need such complex logic